File: test-data/python/relationships/composition.py
class Engine:
    def start(self):
        print("Engine started")

class Wheel:
    def rotate(self):
        print("Wheel rotating")

class Car:
    def __init__(self):
        self.engine = Engine()
        self.wheels = [Wheel() for _ in range(4)]

    def drive(self):
        self.engine.start()
        for wheel in self.wheels:
            wheel.rotate()

File: test-data/typescript/relationships/composition.ts
class Engine {
    start() {
        console.log("Engine started");
    }
}

class Wheel {
    rotate() {
        console.log("Wheel rotating");
    }
}

export class Car {
    private engine = new Engine();
    private wheels: Wheel[] = [new Wheel(), new Wheel(), new Wheel(), new Wheel()];

    drive() {
        this.engine.start();
        this.wheels.forEach(w => w.rotate());
    }
}
